engine/apps/slack/scenarios/distribute_alerts.py

```diff
@@ -128,17 +129,29 @@ def process_signal(self, alert: Alert) -> None:
         if num_updated_rows == 1:
             # this will be the case in the event that we haven't yet created a Slack message for this alert group
 
-            slack_channel = (
-                alert_group.channel_filter.slack_channel
-                if alert_group.channel_filter
-                # if channel filter is deleted mid escalation, use default Slack channel
-                else alert_receive_channel.organization.default_slack_channel
-            )
+            # if channel filter is deleted mid escalation, use the organization's default Slack channel
+            slack_channel = channel_filter.slack_channel if channel_filter else organization.default_slack_channel
+
+            # slack_channel can be None if
+            # - the channel filter is deleted mid escalation
+            # - AND the organization does not have a default slack channel set
+            # in this case, we have to skip posting to Slack because we don't know where to post to
+            if slack_channel is None:
+                logger.info(
+                    f"Skipping posting message to Slack for alert_group {alert_group_pk} because we don't know which "
+                    f"Slack channel to post to. channel_filter={channel_filter} "
+                    f"organization.default_slack_channel={organization.default_slack_channel}"
+                )
+
+                alert_group.slack_message_sent = False
+                alert_group.reason_to_skip_escalation = AlertGroup.CHANNEL_NOT_SPECIFIED
+                alert_group.save(update_fields=["slack_message_sent", "reason_to_skip_escalation"])
+                return
```

Comment:
main change (TODO: comment why this is like this)

Reply:
going to try reproducing this theory locally just to be sure. Since I've been mucking around a lot in this file lately, decided to add more comprehensive test coverage in [engine/apps/slack/tests/scenario_steps/test_distribute_alerts.py](https://github.com/grafana/oncall/pull/5314/files#diff-4878c162db9a602b78ec80d84d36d1e936f12bad0b26a02b40d7b1910c9e92cb)